Python/import.c

```diff
@@ -699,11 +836,12 @@ static void
 remove_module(PyObject *name)
 {
     PyObject *modules = PyImport_GetModuleDict();
-    if (PyDict_GetItem(modules, name) == NULL)
-        return;
-    if (PyDict_DelItem(modules, name) < 0)
+    if (PyMapping_DelItem(modules, name) < 0) {
+        if (!PyMapping_HasKey(modules, name))
```

Comment:
I would prefer to first test PyMapping_HasKey(), as done in the current code.

Reply:
I re-ordered it because it's slightly faster.  However, it's not much of a difference and I don't mind changing it back.